tests/integration/test_transformers.py

```diff
@@ -0,0 +1,188 @@
+    """Test the given transformer with the given input data.
+
+    This method verifies the transformed data's dtype, the reverse
+    transformed data (if `is_composition_identity`) and the dtype.
+
+    Args:
+        transformer_class (rdt.transformers.BaseTransformer):
+            The transformer class to test.
+        input_data (pandas.Series):
+            The data to test on.
+    """
+    transformer = transformer_class()
+    # Fit
+    transformer.fit(input_data, [TEST_COL])
+
+    # Transform
+    transformed = transformer.transform(input_data)
+    _validate_transformed_data(transformer, transformed)
+
+    # Reverse transform
+    out = transformer.reverse_transform(transformed)
+    _validate_reverse_transformed_data(transformer, out, input_data.dtypes[TEST_COL])
+
+    _validate_composition(transformer, out, input_data)
+
+
+def _validate_hypertransformer_transformed_data(transformed_data):
+    """Check that the transformed data is of type float."""
+    for dt in transformed_data.dtypes:
+        assert dt.kind in DATA_TYPE_TO_DTYPES['numerical']
+
+
+def _validate_hypertransformer_reversed_transformed_data(transformer, reversed_data):
+    """Check that the reverse transformed data has the same dtype as the input."""
+    expected_data_type = transformer().get_input_type()
+    assert reversed_data.dtype.kind in DATA_TYPE_TO_DTYPES[expected_data_type]
+
+
+def _test_transformer_with_hypertransformer(transformer_class, input_data):
+    """Test the given transformer in the hypertransformer.
+
+    Run the provided transformer using the hypertransformer using the provided
+    input data. Verify that the expected dtypes are returned by transform
+    and reverse_transform.
+
+    Args:
+        transformer_class (rdt.transformers.BaseTransformer):
+            The transformer class to test.
+        input_data (pandas.Series):
+            The data to test on.
+    """
+    hypertransformer = HyperTransformer(transformers={
+        TEST_COL: {'class': transformer_class.__name__},
+    })
+    hypertransformer.fit(input_data)
+
+    transformed = hypertransformer.transform(input_data)
+    _validate_hypertransformer_transformed_data(transformed)
+
+    out = hypertransformer.reverse_transform(transformed)
+    _validate_hypertransformer_reversed_transformed_data(transformer_class, out[TEST_COL])
+
+
+transformers = BaseTransformer.get_subclasses()
+generators = _build_generator_map()
+
+
+@pytest.mark.parametrize('transformer', transformers)
+def test_transformer(subtests, transformer):
+    """Test the transformer end-to-end.
+
+    Test the transformer end-to-end with at least one generated dataset. Test
+    both the transformer by itself, and by running in the hypertransformer.
+
+    Args:
+        transformer (rdt.transformers.BaseTransformer):
+            The transformer to test.
+    """
+    input_data_type = transformer.get_input_type()
+    assert input_data_type is not None
```

Comment:
I think we can skip this validation, and in the `_find_dataset_generators` function below return all the dataset generators that exist.